src/controllers/userController.js

```diff
@@ -55,22 +56,22 @@ class UserController {
     try {
       const { rows } = await db.query(queryUsersByEmail, params);
       if (rows) {
-        if (rows[0]) {
-          const comparePassword = compareSync(req.body.password, rows[0].password);
-          if (comparePassword) {
-            const authUser = rows[0];
-            const token = generateToken(authUser);
-            return res.status(200).json({
-              status: 200,
-              data: [token],
-            });
-          }
-          if (!comparePassword) {
-            return res.status(401).json({
-              status: 401,
-              error: 'Incorrect login details',
-            });
-          }
+        const comparePassword = compareSync(req.body.password, rows[0].password);
+        if (comparePassword) {

```

Comment:
Always checkout for errors first. You could rewrite this as:

```
if (!comparePassword) {
   return res.status(401).json({
     status: 401,
      error: 'Incorrect login details',
   });
 }
const authUser = rows[0];
   const token = generateToken(authUser);
   return res.status(200).json({
     status: 200,
     data: [{
        token,
      }],
   });
 }

```

Reply:
ok..I will check across all and make the necessary changes

Reply:
changed. Thanks